### DISCUSSION TITLE
Help with Deploying taskiq-faststream App on Any Google Deployment Services

### DISCUSSION THREAD

**Discussion by Atanusaha143** [taskiq-python/taskiq#385] [Q&A]
Hi everyone,

I'm working on deploying a [taskiq-faststream](https://github.com/taskiq-python/taskiq-faststream) app on Google Cloud Run but running into some issues. However, Cloud Run isn't accepting my container image because it’s not starting or listening on the expected port.

Has anyone successfully deployed a [taskiq-faststream](https://github.com/taskiq-python/taskiq-faststream)-based app on Cloud Run or in any other Google service? I’d appreciate any guidance or troubleshooting tips. 

Thanks!

**Comment by s3rius** [comment]
Implementation is platform independent, so I guess deploying application as you usually on GCP should be enough. @Lancetnik might have examples of using faststream on GCP.

**Reply by Lancetnik** [reply]
It is easy to use with `AsgiFastStream`, but I am not sure, how it should works with taskiq executor. Can we make it listen specific HTTP port?

**Reply by s3rius** [reply]
I believe you can run it with distributed queues as usual; HTTP isn’t required in this case. Most likely, it will just spin up some machines where you can deploy your application, at least according to the Cloud Run documentation.

@Atanusaha143, it also seems like you can run Docker images on Cloud Run, so packaging your app in Docker should be sufficient for that. At least that’s my understanding.

Sorry, I’m not a big fan of GCP and don’t have extensive knowledge of its services. But if you have GKE or EKS, it would be pretty much the same as deploying on Kubernetes.
